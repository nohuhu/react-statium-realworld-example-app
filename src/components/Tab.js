import React from 'react';
import { useStore } from 'statium';
import { Link } from 'react-router-dom';

const Tab = ({ id, name, to, currentTab, setTab }) => {
  const { dispatch } = useStore();

  const cls = `nav-link ${id === currentTab ? 'active' : ''}`;

  return (
    <li className="nav-item">
      <Link to="" className={cls} onClick={() => dispatch(setTab, to)}>
        {name}
      </Link>
    </li>
  );
}

export default Tab;
