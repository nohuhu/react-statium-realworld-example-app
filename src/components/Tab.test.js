import React from 'react';
import Store from 'statium';
import { mount } from 'test/enzyme.js';

import Tab from './Tab.jsx';

// We need a Store to dispatch actions. Nothing special needed, just an empty Store.
const TestTab = props => (
  <Store tag="test">
    <Tab {...props} />
  </Store>
);

describe("Tab widget", () => {
  it("should render with inactive cls when currentTab !== id", () => {
    const tree = mount(<TestTab id="foo" name="Foo tab" to="/foo" currentTab="bar" />);

    expect(tree.find('Tab')).toMatchSnapshot();
    expect(tree.find('Link').hasClass('active')).toBe(false);
  });

  it("should render with active cls when currentTab === id", () => {
    const tree = mount(<TestTab id="bar" name="Bar tab :)" to="/bar" currentTab="bar" />);

    expect(tree.find('Tab')).toMatchSnapshot();
    expect(tree.find('Link').hasClass('active')).toBe(true);
  });

  it("should dispatch the passed setTab function on click", async () => {
    const spy = jest.fn();

    const tree = mount(<TestTab id="baz" name="Baz tab" to="/baz" currentTab="bar" setTab={spy} />);

    tree.find('Link').simulate('click');

    // Give the action enough time to fire
    await sleep(10);

    expect(spy).toHaveBeenCalled();

    const args = spy.mock.calls[0];

    // First argument is Store API: { data, state, set, dispatch }. We don't need to test it.
    expect(args[1]).toBe("/baz");
  });
});
