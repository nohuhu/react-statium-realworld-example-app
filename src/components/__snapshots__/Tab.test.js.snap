// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab widget should render with active cls when currentTab === id 1`] = `
<Tab
  currentTab="bar"
  id="bar"
  name="Bar tab :)"
  to="/bar"
>
  <li
    className="nav-item"
  >
    <Link
      className="nav-link active"
      onClick={[Function]}
      to=""
    >
      <a
        className="nav-link active"
        href="/"
        onClick={[Function]}
      >
        Bar tab :)
      </a>
    </Link>
  </li>
</Tab>
`;

exports[`Tab widget should render with inactive cls when currentTab !== id 1`] = `
<Tab
  currentTab="bar"
  id="foo"
  name="Foo tab"
  to="/foo"
>
  <li
    className="nav-item"
  >
    <Link
      className="nav-link "
      onClick={[Function]}
      to=""
    >
      <a
        className="nav-link "
        href="/"
        onClick={[Function]}
      >
        Foo tab
      </a>
    </Link>
  </li>
</Tab>
`;
